qlty-cli/src/commands/coverage/validate.rs

```diff
@@ -0,0 +1,43 @@
+use crate::{CommandError, CommandSuccess};
+use anyhow::Result;
+use clap::Args;
+use qlty_coverage::validate::{ValidationResult, ValidationStatus};
+use serde_json;
+
+#[derive(Debug, Args)]
+pub struct Validate {
+    #[arg(long)]
+    /// The minimum percentage of coverage report files to match the file system for validation to succeed.
+    /// If not specified, defaults to 90%.
+    pub threshold: Option<f64>,
+
+    // Path to zip file
+    pub path: String,
+
+    // Print the result in JSON format
+    #[arg(long, default_value_t = false)]
+    pub json: bool,
+}
+
+impl Validate {
+    pub fn execute(&self, _args: &crate::Arguments) -> Result<CommandSuccess, CommandError> {
+        let validation_result = ValidationResult::compute(&self.path, self.threshold)?;
+
+        if self.json {
+            println!("{}", serde_json::to_string_pretty(&validation_result)?);
+        }
+
+        match validation_result.status {
```

Comment:
We have this concept of transforming a validation result into a command result in a couple places

I think we can refactor this into a trait implementation of the `From` trait

Reply:
That would be something like https://github.com/qltysh/qlty/pull/1915/commits/e13c25454e293156c99b1e21a5902c59a2ed85c1?

But https://github.com/qltysh/qlty/pull/1915/commits/9b6261256ba7facd895975889e66db21059f273a might be simpler.